azafea/event_processors/endless/metrics/request.py

```diff
@@ -34,6 +34,13 @@ class Request(Base):
     send_number = Column(Integer, nullable=False)
 
 
+class RequestByDay(View):
+    __tablename__ = 'metrics_request_v2_by_day'
+    __query__ = DbSession().query(
+        Request.received_at.cast(Date).distinct().label('day'),
```

Comment:
My sqlalchemy is rusty but are we sure this evaluates to `SELECT DISTINCT received_at::date, machine_id` rather than `SELECT DISTINCT ON (received_at :: date) day, machine_id`, which would mean "for each distinct date, pick an arbitrary machine ID".

Reply:
You’re right. The query is actually only used (for the moment) to get the tables that the view needs before it is created. The "real" request is in the deployment, and it’s not generated from this query (yet). It’s now fixed.